logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/server/CustomServerConfigRepository.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.configuration.server
+
+import com.benasher44.uuid.uuid4
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.di.MapperProvider
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.logic.wrapStorageRequest
+import com.wire.kalium.network.BackendMetaDataUtil
+import com.wire.kalium.network.BackendMetaDataUtilImpl
+import com.wire.kalium.network.api.base.unbound.configuration.ServerConfigApi
+import com.wire.kalium.network.api.base.unbound.versioning.VersionInfoDTO
+import com.wire.kalium.persistence.daokaliumdb.ServerConfigurationDAO
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+internal interface CustomServerConfigRepository {
```

Comment:
it was not to solve TooManyFunctions only but also to separate 2 uses that class was covering:
1. Call wire server for some stuff like API version and blacklist ect
2. fetch a Server config from a deep link

and we need to separate those 2 since any thing that is wire server related MUST be called with the proxy if there is a proxy setup

Reply:
Makes sense! Sounds good.

`CustomServer` is a term we haven't used in the codebase yet, I believe. So that's why I thought of clarifying what it is and what goes where.